raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -53,24 +57,55 @@
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
+@RunWith(Parameterized.class)
 public class KafkaRaftClientTest {
     private final int localId = 0;
     private final int electionTimeoutMs = 10000;
     private final int retryBackoffMs = 50;
     private final int requestTimeoutMs = 5000;
     private final int electionJitterMs = 0;
     private final MockTime time = new MockTime();
-    private final MockElectionStore electionStore = new MockElectionStore();
     private final MockLog log = new MockLog();
     private final MockNetworkChannel channel = new MockNetworkChannel();
 
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<String[]> data() {
```

Comment:
Hmm, I think integration testing should be done separately.

Reply:
Could you elaborate?

Reply:
The benefit of a unit test is that it can isolate the behavior of a specific component. When we start mixing in the real implementations, we lose that isolation. We don't have to be too rigid about this, but I do think there is value in keeping test cases simple. For example, when we start adding implementations of additional components, are we going to be testing every permutation? Would that even be useful? My slight preference here is to make this more of a pure unit test and create a separate integration test. Then we can also provide the real network/log implementations.

Reply:
I see the point.